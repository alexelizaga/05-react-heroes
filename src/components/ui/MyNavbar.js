import { Navbar, Nav, Container } from 'react-bootstrap';
import { Link } from 'react-router-dom';

export const MyNavbar = () => {
  return (
    <Navbar bg='dark' variant='dark' expand='sm'>
      <Container>
        <Navbar.Brand as={Link} to="/">Asociaciones</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            <Nav.Link as={Link} to="/marvel">Marvel</Nav.Link>
            <Nav.Link as={Link} to="/dc">DC</Nav.Link>
          </Nav>
        </Navbar.Collapse>

        <Navbar.Collapse className="justify-content-end">
          <Nav>
            <Nav.Link as={Link} to="/login">Logout</Nav.Link>
          </Nav>
          
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};
