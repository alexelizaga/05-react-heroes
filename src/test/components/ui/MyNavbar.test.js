import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

import { MyNavbar } from '../../../components/ui/MyNavbar';
import { AuthContext } from '../../../auth/authContext';
import { types } from '../../../types/types';

const mockNavigate = jest.fn();
const mockDispatch = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate
}));

describe('Test <MyNavbar />', () => {

  const contextValue = {
    user: {
      logged: true,
      name: 'Pedro'
    },
    dispatch: mockDispatch
  };

  test('should render correctly', () => {
    const view = render(
      <AuthContext.Provider value={ contextValue }>
        <MemoryRouter initialEntries={['/']}>
          <Routes>
            <Route path='/' element={<MyNavbar />} />
          </Routes>
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(view.baseElement).toMatchSnapshot();

    const myNavbarUserName = screen.queryByTestId('MyNavbarUserName');
    expect(myNavbarUserName.textContent.trim()).toBe(contextValue.user.name);
  });

  test('should handle logout', () => {
    render(
      <AuthContext.Provider value={ contextValue }>
        <MemoryRouter initialEntries={['/']}>
          <MyNavbar />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    const myNavbarLogoutBtn = screen.queryByTestId('MyNavbarLogoutBtn');
    fireEvent.click(myNavbarLogoutBtn);

    expect(mockDispatch).toBeCalledWith({ 'type': types.logout });
    expect(mockNavigate).toBeCalledWith('/login', { replace: true });
  });

});