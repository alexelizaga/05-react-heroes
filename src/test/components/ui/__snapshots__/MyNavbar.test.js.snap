// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <MyNavbar /> should render correctly 1`] = `
<body>
  <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
      data-testid="MyNavbar"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Asociaciones
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/marvel"
              href="/marvel"
            >
              Marvel
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/dc"
              href="/dc"
            >
              DC
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/search"
              href="/search"
            >
              Search
            </a>
          </div>
        </div>
        <div
          class="justify-content-end navbar-collapse collapse"
        >
          <span
            class="text-info navbar-text"
            data-testid="MyNavbarUserName"
          >
            Pedro
          </span>
          <div
            class="navbar-nav"
          >
            <button
              class="btn btn-dark"
              data-testid="MyNavbarLogoutBtn"
              type="button"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>
</body>
`;
